x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -239,12 +248,13 @@ export const ComplianceDashboard = () => {
         label: i18n.translate('xpack.csp.dashboardTabs.cloudTab.tabTitle', {
           defaultMessage: 'Cloud',
         }),
+        'data-test-subj': CLOUD_DASHBOARD_TAB,
         isSelected: selectedTab === CSPM_POLICY_TEMPLATE,
         onClick: () => setSelectedTab(CSPM_POLICY_TEMPLATE),
         content: (
           <>
-            {hasFindingsCspm ? (
-              <CloudPosturePage query={getCspmDashboardData}>
+            {hasFindingsCspm || !isCspmInstalled ? (
+              <CloudPosturePage query={getCspmDashboardData} hasFindings={hasFindingsCspm}>
```

Comment:
We should call CloudPosturePage only once for the dashboard page (we are already wrapping it on line 322),

Let's make CloudPosturePage concerns only about if any of the integrations were installed or not, and Instead of calling the again CloudPosture page component for the tabs, Let's create separate components for Cloud and Kspm tabs and use the function I suggested [here](https://github.com/elastic/kibana/pull/150134/files#r1153605917) to determine what to render.

Reply:
@CohenIdo let me know your thoughts on that, maybe that can be addressed in a  follow up PR